chapter_computational-performance/parameterserver_vn.md

```diff
@@ -81,13 +85,22 @@ The bandwidth from the CPU on a 16x Gen3 link is 16GB/s.
 This is also the speed at which *each* of the GPUs is connected to the switch. This means that it is more effective to communicate between the
 -->
 
-*dịch đoạn phía trên*
+Cách lý luận này có thể rất tùy tiện và phù phiếm.
+Sau cùng thì, phần toán xuyên suốt bên dưới vẫn không thay đổi.
+Nhưng ở đây, chúng ta đang làm việc với các thiết bị phần cứng vật lý với các bus có những băng thông khác nhau như đã thảo luận trong :numref:`sec_hardware`.
+Xét một máy chủ GPU 4-chiều như miêu tả trong :numref:`fig_bw_hierarchy`.
+Nếu nó được kết nối cực kỳ tốt, nó có thể sở hữu một card mạng với 100 GbE.
+Con số chi tiết hơn là ở trong khoảng 1-10 GbE với băng thông hiệu dụng từ 100MB/s đến 1GB/s.
+Vì các CPU thường có quá ít làn PCIe để kết nối với toàn bộ GPU một cách trực tiếp
+(ví dụ, CPU thông dụng của Intel có 24 làn) ta cần một [multiplexer](https://www.broadcom.com/products/pcie-switches-bridges/pcie-switches) (mạch đa hợp, mạch dồn kênh).
+Băng thông của một CPU với một 16x Gen3 là 16GB/s.
+Đây cũng là tốc độ mà *mỗi* GPU được kết nối với bộ chuyển mạch. Điều này có nghĩa là việc truyền tin giữa CPU và các GPU sẽ hiệu quả hơn.
```

Comment:
```suggestion
Đây cũng là tốc độ mà *mỗi* GPU được kết nối với bộ chuyển mạch. Điều này có nghĩa là việc truyền tin trực tiếp giữa các GPU sẽ hiệu quả hơn.
```
bản gốc có thêm chữ `devices` ở cuối rồi.

Reply:
`devices` này em thấy cũng có thể là bao gồm CPU luôn chứ

Reply:
Chỗ này a hiểu theo đoạn dưới ấy, ví dụ như aggregate gradient trên cpu, truyền từ 4 GPU về CPU sẽ chậm hơn, tốn 80ms , trong khi truyền về aggregate theo GPU0 thì tốn 30ms, còn chia đều truyền trên tất cả GPU thì chỉ tốn 15sms.
Vì băng thông PCIe tới CPU là 16GB, mà mỗi GPU đều có một bus PCIe 16GB, vậy truyền từ cả 4 GPUs về CPU thì sẽ không thể sử dụng toàn bộ 16GB x 4 băng thông từ tất cả GPU được. Vậy nên truyền giữa GPUs mới hiệu quả **hơn**.

Reply:
thanks anh @lkhphuc đã thông não 💯. Quên mất đoạn phía trên đang triển khai ý so sánh tổng hợp gradient trên đâu là nhanh nhất.